spec/requests/api/create_article_spec.rb

```diff
@@ -1,11 +1,14 @@
 RSpec.describe 'POST /api/articles', type: :request do
   subject { response }
+  let(:category) { create(:category) }
   describe 'successful, when the article is created' do
     before do
       post '/api/articles',
            params: { article: { title: 'Amazing title',
                                 lede: 'Amazing lede...',
-                                body: 'Amazing body' } }
+                                body: 'Amazing body',
+                                category_id: category.id,
+                                category_name: category.name } }
```

Comment:
I would say it is a redundant and complicated to set both category :id and :name in params. That couples front end with back end which is bad, they should be as decoupled as possible. Lets say you changed something in back end that changes the id of category `tech`. Front end need to know about it and store it, which is not its job at all. It only need to filter articles by category and that's it. Try to make code so that sending only category name in params is sufficient for your functionality.

Reply:
It is more time now to fix it, but it will save you time on fixing bugs later

Reply:
This was a good challenge. We've implemented a helper function in `articles_controller` to look up the correct `category_id` from the `categories` table, and so we're able to remove the `category_id` param for creating a new article as you suggest.